PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -832,4 +842,42 @@ bool str2Clipboard(const generic_string &str2cpy, HWND hwnd)
 		return false;
 	}
 	return true;
-}
\ No newline at end of file
+}
+
+// string trimming; algorithms taken from http://stackoverflow.com/questions/216823/whats-the-best-way-to-trim-stdstring
+generic_string stringTrimLeft(const generic_string & s)
+{
+	generic_string result(s);
+	result.erase(result.begin(), std::find_if(result.begin(), result.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+	return result;
+}
+
+generic_string stringTrimRight(const generic_string & s)
+{
+	generic_string result(s);
+	result.erase(std::find_if(result.rbegin(), result.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), result.end());
+	return result;
+}
+
+generic_string & stringTrimLeftInline(generic_string & result)
+{
+	result.erase(result.begin(), std::find_if(result.begin(), result.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+	return result;

```

Comment:
What's the point of *returning* the reference? I can't imagine that this is actually good for anything other than making alias analysis harder for the compiler.